scene/resources/material.cpp

```diff
@@ -454,6 +454,20 @@ void ShaderMaterial::set_shader_parameter(const StringName &p_param, const Varia
 	} else {
 		Variant *v = param_cache.getptr(p_param);
 		if (!v) {
+#ifdef TOOLS_ENABLED
+			bool missing;
+			if (param_cache.is_empty()) {
+				// Build parameter cache.
+				List<PropertyInfo> l;
+				_get_property_list(&l);
+				missing = !param_cache.has(p_param);;
+			} else {
+				missing = true;
+			}
+			if (missing) {
+				WARN_PRINT(vformat("Shader parameter \"%s\" not found in Shader.", p_param));
+			}
```

Comment:
Suggested style tweak.

I haven't assessed whether the solution is acceptable here in Material.
```suggestion
			bool missing = true;
			if (param_cache.is_empty()) {
				// Build parameter cache.
				List<PropertyInfo> l;
				_get_property_list(&l);
				missing = !param_cache.has(p_param);
			}
			if (missing) {
				WARN_PRINT(vformat("Shader parameter \"%s\" not found in Shader.", p_param));
			}
```